### PR TITLE
Config parser

### PR DESCRIPTION
**Pull request by mackan1410**
Functions for parsing xlsx-files, and an interface for uploading these files.
When a file is uploaded, it is parsed, and the result is stored in localstorage.

### CODE REVIEW

File: src/components/diagram/diagram.js [hunk 1]
```diff
@@ -199,6 +199,7 @@ const Diagram = ({ fileName }) => {
       children: result.nodes,
       edges: result.edges
     })
+    .catch(abort)
```
Comment (JoXW100) [inline src/components/diagram/diagram.js:202]:
```suggestion
```
remove catch from awaited promise

File: src/components/diagram/diagram.js [hunk 2]
```diff
@@ -227,6 +228,12 @@ const Diagram = ({ fileName }) => {
     }
   }, [fileName, data])
 
+  const abort = () => {
+    console.log("abort")
+    localStorage.removeItem('diagram')
+    window.location.href = window.location.href
+  }
+
```
Comment (JoXW100) [inline src/components/diagram/diagram.js:236]:
```suggestion
```
Remove unused function

**Review by JoXW100 — CHANGES_REQUESTED** [review]

**Review by JoXW100 — APPROVED** [review]

File: src/pages/index.js [hunk 3]
```diff
@@ -1,22 +1,65 @@
...
-      <link rel='icon' href='/favicon.ico' />
-    </Head>
-
-    <main className={styles.main}>
-      <DiagramContext data={require('../data/example.json')} />
-    </main>
-  </div>
-)
+const Main = () => {
+  const [state, setState] = useState([])
```
Comment (thomssonkarl) [inline src/pages/index.js:7]: Change this in the future to be more readable

File: src/pages/upload.js [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    const parse = require('../utils/parse/parse')
+
+    const reader = new FileReader()
+    reader.onload = (event) => {
+      const data = event.target.result
+
+      try {
+        const graphMap = parse.parseXLSX(data)
+        localstorage.setDiagram(graphMap.toJson())
+        window.location.href = 'http://localhost:3000'
```
Comment (thomssonkarl) [inline src/pages/upload.js:21]: This Will break upon deployment

**Review by thomssonkarl — APPROVED** [review]

**Comment by thomssonkarl** [comment]
Some things should probably be changed before deployment but cant really be adressed atm

**Pull request merged**